arangod/VocBase/vocbase.h

```diff
@@ -167,49 +176,43 @@ struct TRI_vocbase_t {
-      arangodb::replication2::LogId id) const
-      -> std::shared_ptr<arangodb::replication2::replicated_log::LogFollower>;
-  [[nodiscard]] auto getReplicatedLogs() const
-      -> std::unordered_map<arangodb::replication2::LogId,
-                            arangodb::replication2::replicated_log::LogStatus>;
-  [[nodiscard]] auto getReplicatedLogsQuickStatus() const -> std::unordered_map<
-      arangodb::replication2::LogId,
-      arangodb::replication2::replicated_log::QuickLogStatus>;
-  [[nodiscard]] auto createReplicatedLog(
-      arangodb::replication2::LogId id,
-      std::optional<std::string> const& collectionName)
-      -> arangodb::ResultT<std::shared_ptr<
-          arangodb::replication2::replicated_log::ReplicatedLog>>;
-  [[nodiscard]] auto dropReplicatedLog(arangodb::replication2::LogId id)
-      -> arangodb::Result;
-  auto ensureReplicatedLog(arangodb::replication2::LogId id,
-                           std::optional<std::string> const& collectionName)
-      -> std::shared_ptr<arangodb::replication2::replicated_log::ReplicatedLog>;
 
  public:
   auto createReplicatedState(arangodb::replication2::LogId id,
-                             std::string_view type)
+                             std::string_view type,
+                             arangodb::velocypack::Slice parameter)
       -> arangodb::ResultT<std::shared_ptr<
           arangodb::replication2::replicated_state::ReplicatedStateBase>>;
-  auto dropReplicatedState(arangodb::replication2::LogId id)
+  auto dropReplicatedState(arangodb::replication2::LogId id) noexcept
       -> arangodb::Result;
-  auto ensureReplicatedState(arangodb::replication2::LogId id,
-                             std::string_view type)
-      -> std::shared_ptr<
-          arangodb::replication2::replicated_state::ReplicatedStateBase>;
-  [[nodiscard]] auto getReplicatedStateStatus() const -> std::unordered_map<
-      arangodb::replication2::LogId,
-      std::optional<arangodb::replication2::replicated_state::StateStatus>>;
-  [[nodiscard]] auto getReplicatedStateById(arangodb::replication2::LogId id)
-      const -> std::shared_ptr<
-          arangodb::replication2::replicated_state::ReplicatedStateBase>;
+
+  auto getReplicatedStateById(arangodb::replication2::LogId id)
+      -> arangodb::ResultT<std::shared_ptr<
+          arangodb::replication2::replicated_state::ReplicatedStateBase>>;
+
+  auto updateReplicatedState(
+      arangodb::replication2::LogId id,
+      arangodb::replication2::agency::LogPlanTermSpecification const& spec,
+      arangodb::replication2::agency::ParticipantsConfig const&)
+      -> arangodb::Result;
+
+  [[nodiscard]] auto getReplicatedStatesQuickStatus() const
+      -> std::unordered_map<
+          arangodb::replication2::LogId,
+          arangodb::replication2::replicated_log::QuickLogStatus>;
+
+  [[nodiscard]] auto getReplicatedStatesStatus() const
+      -> std::unordered_map<arangodb::replication2::LogId,
```

Comment:
We don't really need FlatHashMap. :)
It just doesn't matter here, so we're sticking with the STL containers for now. Changing to another container is a micro-optimization that's far from useful in the current state, and I'd rather do only when the code is much more stable (stable in the "isn't going to change much any more" sense).

Reply:
The main issue to unordered map (expect it's slower): it gives a lot of commonly unnecessary guarantees.
And in future changing it will be more difficult, because needs to understand is this guarantees necessary or not?

Reply:
> I can still spot that will actually be noticeably is that we currently have no pretty printer for abseil, making it annoying to debug. 

Good point, will be fixed!

But as I wrote it's here not for making you immediately replace it, just for start discussion

Reply:
Anyway I think it's need to be resolved.

Reply:
If you want to continue this discussion, shall we move it to DM?